gnssanalysis/gn_download.py

```diff
@@ -846,325 +848,169 @@ def download_multiple_files_from_cddis(files: [str], ftp_folder: str, output_fol
-                dates=dt_list, dest=dest, ac=ac, f_type=ac_typ_dict[ac], dwn_src="cddis", f_dict=True, repro3=True
-            )
-        else:
-            f_dict_update = download_prod(
-                dates=dt_list, dest=dest, ac=ac, f_type=ac_typ_dict[ac], dwn_src="cddis", f_dict=True
-            )
-        f_dict.update(f_dict_update)
-
-    f_types = []
-    for el in list(ac_typ_dict.values()):
-        for typ in el:
-            f_types.append(typ)
-    if most_recent:
-        f_types.append("snx")
-
-    # Prepare the output dictionary based on the downloaded files:
-    for f_type in f_types:
-        if f_type == "rnx":
-            out_dict[f"navfiles"] = sorted(f_dict[f_type])
-        out_dict[f"{f_type}files"] = sorted(f_dict[f_type])
+        atx_filename = "igs14.atx"
+    ensure_folders([download_dir])
+    url = f"https://files.igs.org/pub/station/general/{atx_filename}"
+    attempt_url_download(
+        download_dir=download_dir, url=url, filename=atx_filename, type_of_file="ATX", if_file_present=if_file_present
+    )
 
-    return out_dict
 
+def download_satellite_metadata_snx(download_dir: _Path, if_file_present: str = "prompt_user") -> _Path:
+    """Download the most recent IGS satellite metadata file
 
-def download_rinex3(dates, stations, dest, dwn_src="cddis", ftps=False, f_dict=False):
-    """
-    Function used to get the RINEX3 observation file from download server of choice, default: CDDIS
+    :param _Path download_dir: Directory to store the file in
+    :param str if_file_present: What to do if file already present: "replace", "dont_replace", defaults to "prompt_user"
+    :return _Path: Return the pathlib.Path of the downloaded file
     """
-    if dest[-1] != "/":
-        dest += "/"
-    # Convert input to list of datetime dates (if not already)
-    dt_list = dates_type_convert(dates)
+    ensure_folders([download_dir])
+    download_filepath = attempt_url_download(
+        download_dir=download_dir,
+        url="https://files.igs.org/pub/station/general/igs_satellite_metadata.snx",
+        filename="igs_satellite_metadata.snx",
+        type_of_file="IGS satellite metadata",
+        if_file_present=if_file_present,
+    )
+    return download_filepath
 
-    if isinstance(stations, str):
-        stations = [stations]
 
-    # Create directory if doesn't exist:
-    if not _Path(dest).is_dir():
-        _Path(dest).mkdir(parents=True)
+def download_yaw_files(download_dir: _Path, if_file_present: str = "prompt_user") -> _List[_Path]:
+    """_summary_
 
-    if f_dict:
-        f_dict = {"rnxfiles": []}
+    :param _Path download_dir: Directory to store the file in
+    :param str if_file_present: What to do if file already present: "replace", "dont_replace", defaults to "prompt_user"
+    :return _List[_Path]: _description_
+    """
+    ensure_folders([download_dir])
+    urls = [
+        "https://peanpod.s3.ap-southeast-2.amazonaws.com/aux/products/tables/bds_yaw_modes.snx.gz",
```

Comment:
We should potentially be constructing this from a config class or something, rather than hard-coding the full value.

Reply:
Put in base URLs at top of file

Reply:
And other ftp urls as well? So we can use other ftp/https sites in case the current one is down.